linkerd/app/outbound/src/discover.rs

```diff
@@ -68,109 +72,181 @@ pub(crate) fn resolver(
-                    policy::BackendDispatcher::Forward(addr.0, Default::default()),
                 );
-                tracing::debug!(
-                    ?policy,
-                    "Policy not found; no profile: synthesizing policy from original destination address"
-                );
-
-                let (tx, policy_rx) = watch::channel(policy);
-                // keep the watch channel alive as long as it's in use.
-                tokio::spawn(async move { tx.closed().await });
-                return Ok((None, policy_rx));
+                return Ok((Some(profile), policy));
             }
 
-            // Otherwise, errors resolving the policy are considered fatal.
-            Err(policy_error)
+            // Otherwise, route the request to the original destination address.
+            let policy = spawn_synthesized_origdst_policy(orig_dst, queue, detect_timeout);
+            Ok((None, policy))
         })
     })
 }
 
+#[inline]
+fn is_not_found(e: &Error) -> bool {
+    e.downcast_ref::<tonic::Status>()
+        .map(|s| s.code() == tonic::Code::NotFound)
+        .unwrap_or(false)
+}
+
+fn spawn_synthesized_profile_policy(
+    orig_dst: SocketAddr,
+    mut profile: watch::Receiver<profiles::Profile>,
+    queue: policy::Queue,
+    detect_timeout: Duration,
+) -> watch::Receiver<policy::ClientPolicy> {
+    let mk = move |profile: &profiles::Profile| {
+        let (addr, meta) = profile
+            .endpoint
+            .clone()
+            .unwrap_or_else(|| (orig_dst, Default::default()));
+        let forward = policy::BackendDispatcher::Forward(addr, meta);
+        synthesize_forward_policy(detect_timeout, queue, forward)
+    };
+
+    let policy = mk(&*profile.borrow_and_update());
+    tracing::debug!(?policy, profile = ?*profile.borrow(), "Synthesizing policy from profile");
+    let (tx, rx) = watch::channel(policy);
+    tokio::spawn(async move {
+        let mut profile = profile;
+        loop {
+            if profile.changed().await.is_err() {
+                tracing::debug!("Profile watch closed, terminating");
+                return;
+            };
+            let policy = mk(&*profile.borrow());
+            tracing::debug!(?policy, "Profile updated; synthesizing policy");
+            if tx.send(policy).is_err() {
+                tracing::debug!("Policy watch closed, terminating");
+                return;
+            }
+        }
+    });
+    rx
+}
+
+fn spawn_synthesized_origdst_policy(
+    orig_dst: SocketAddr,
+    queue: policy::Queue,
+    detect_timeout: Duration,
+) -> watch::Receiver<policy::ClientPolicy> {
+    let forward = policy::BackendDispatcher::Forward(orig_dst, Default::default());
+    let policy = synthesize_forward_policy(detect_timeout, queue, forward);
+    tracing::debug!(?policy, "Synthesizing policy");
+    let (tx, rx) = watch::channel(policy);
+    tokio::spawn(async move {
+        tx.closed().await;
+    });
+    rx
+}
+
+fn synthesize_forward_policy(
```

Comment:
I moved this out of `client-policy` because it's basically a hardcoded configuration. I wouldn't want to rely on it anywhere else but here.